sherpa/sim/tests/test_sim.py

```diff
@@ -129,91 +134,251 @@ def setup():
+def test_uniform_sample2(setup, reset_seed):
+    out = sim.uniform_sample(setup.fit, num=setup.num)
+
+    assert out == pytest.approx(EXPECTED_UNIFORM)
+
+
+def test_normal_sample_vector(setup, reset_seed):
+    ps = sim.NormalSampleFromScaleVector()
+    out = ps.get_sample(setup.fit, num=setup.num)
+
+    assert out == pytest.approx(EXPECTED_NORMAL)
+
+
+def test_normal_sample_matrix(setup, reset_seed):
+    ps = sim.NormalSampleFromScaleMatrix()
+    out = ps.get_sample(setup.fit, num=setup.num)
+
+    assert out == pytest.approx(EXPECTED_NORMAL2)
+
+
+def test_t_sample_matrix(setup, reset_seed):
+    ps = sim.StudentTSampleFromScaleMatrix()
+    out = ps.get_sample(setup.fit, setup.num, setup.dof)
 
+    assert out == pytest.approx(EXPECTED_T)
 
-def test_normal_sample_vector(setup):
-    np = sim.NormalSampleFromScaleVector()
-    np.get_sample(setup.fit, num=setup.num)
 
+def test_normal_sample(setup, reset_seed):
+    out = sim.normal_sample(setup.fit, num=setup.num, correlate=False)
 
-def test_normal_sample_matrix(setup):
-    np = sim.NormalSampleFromScaleMatrix()
-    np.get_sample(setup.fit, num=setup.num)
+    assert out == pytest.approx(EXPECTED_NORMAL)
 
 
-def test_t_sample_matrix(setup):
-    np = sim.StudentTSampleFromScaleMatrix()
-    np.get_sample(setup.fit, setup.num, setup.dof)
+def test_normal_sample_correlated(setup, reset_seed):
+    out = sim.normal_sample(setup.fit, num=setup.num, correlate=True)
 
+    assert out == pytest.approx(EXPECTED_NORMAL2)
 
-def test_normal_sample(setup):
-    sim.normal_sample(setup.fit, num=setup.num, correlate=False)
 
+def test_t_sample(setup, reset_seed):
+    out = sim.t_sample(setup.fit, setup.num, setup.dof)
 
-def test_normal_sample_correlated(setup):
-    sim.normal_sample(setup.fit, num=setup.num, correlate=True)
+    assert out == pytest.approx(EXPECTED_T)
 
 
-def test_t_sample(setup):
-    sim.t_sample(setup.fit, setup.num, setup.dof)
+def test_lrt(setup, reset_seed):
+    """There is a limited check of the results.
 
+    These are regression tests, so will need to be updated if the code
+    or random generator changes. It is actually tricky to test the
+    random numbers as they are going to depend on the number of
+    processes used, and we can not guarantee the number of cores in
+    use (other than forcing it to be 1).
+
+    It looks like each iteration can generate different results
+    depending on the platform (linux vs macOS) due to numerical
+    differences (assumed) - e.g. the fwhm may be 0.03 in one and 0.06
+    in the other, which can then lead to very different amplitudes,
+    even though the resulting fit is not very different.  Then,
+    because the values are not reset after each fit - see issue #1746
+    - we quickly get rather different ratio values.  At least, that is
+    the current hypothesis. So this test is limited to a small number
+    of iterations, where this does difference does not seem to be a
```

Comment:
"does difference does". Remove first does.

Reply:
@hamogu For the MCMC algorithms there are two options - one jumping always from the best location and the other jumping from the new location. There is no fit being done when you run MCMC - just calculation of the likelihood at the point in the parameter space.
plot_pvalue is not an MCMC algorithm to sample parameter space. It performs a simulation and fitting the simulated data at each iteration. The simulated data are from the null model which are then fit with the null and alternative model to calculate likelihood ratio. The algorithm is specified in the last section of Protassov et al paper.

Reply:
I was going to say this is just MC, not MCMC (for whatever you take montecarlo to mean, which I am being very liberal at interpreting here)